config/config-api/src/com/thoughtworks/go/domain/scm/SCM.java

```diff
@@ -198,34 +221,31 @@ public void applyPluginMetadata() {
 
     public void setConfigAttributes(Object attributes) {
         Map attributesMap = (Map) attributes;
-        if (attributesMap.containsKey(NAME)) {
-            name = ((String) attributesMap.get(NAME));
-        }
         if (attributesMap.containsKey(SCM_ID)) {
             id = ((String) attributesMap.get(SCM_ID));
         }
+        if (attributesMap.containsKey(NAME)) {
+            name = ((String) attributesMap.get(NAME));
+        }
+        this.setAutoUpdate("true".equals(attributesMap.get(AUTO_UPDATE)));
         if (attributesMap.containsKey(PLUGIN_CONFIGURATION)) {
             pluginConfiguration.setConfigAttributes(attributesMap.get(PLUGIN_CONFIGURATION));
         }
-        if (attributesMap.containsKey(Configuration.CONFIGURATION)) {
-            configuration.clear();
-            configuration.setConfigAttributes(attributesMap.get(Configuration.CONFIGURATION), getSecureKeyInfoProvider());
-        }
+        setPluginConfigurationAttributes(attributesMap);
     }
 
-    private SecureKeyInfoProvider getSecureKeyInfoProvider() {
-        final SCMMetadataStore scmMetadataStore = SCMMetadataStore.getInstance();
-        final SCMConfigurations metadata = scmMetadataStore.getConfigurationMetadata(getPluginId());
-        if (metadata == null) {
-            return null;
-        }
-        return new SecureKeyInfoProvider() {
-            @Override
-            public boolean isSecure(String key) {
-                SCMConfiguration configuration = metadata.get(key);
-                return configuration.getOption(SCMConfiguration.SECURE);
+    protected void setPluginConfigurationAttributes(Map attributes) {
+        SCMConfigurations scmConfigurations = SCMMetadataStore.getInstance().getConfigurationMetadata(pluginConfiguration.getId());
+        for (SCMConfiguration scmConfiguration : scmConfigurations.list()) {
```

Comment:
The scmConfigurations list can be null. Is not handled. Any reason? It was handled earlier ([```metadata == null```](https://github.com/gocd/gocd/pull/865/files#diff-241b146f9f85defaee538d834fcbe881L219)).